### PR TITLE
Use background color unscoped modifier in box test

### LINKED ISSUE #1583: Add backgroundColor = boxColor once the modifiers are available.
https://api.github.com/cashapp/redwood/blob/9cf3c0c23ccf21b27d6545f9ef87cfd7efeb5936/test-app/presenter/src/commonMain/kotlin/com/example/redwood/testing/presenter/BoxSandbox.kt#L201

```text

}

@Composable
private fun ColumnScope.Legend() {
  Text(
    "Legend",
    modifier = Modifier.horizontalAlignment(Center).height(40.dp),
  )
  Box(
    width = Fill,
    height = Wrap,
    horizontalAlignment = Stretch,
    verticalAlignment = Start,
    modifier = Modifier.height(80.dp).margin(Margin(horizontal = 20.dp)),
  ) {
    Rectangle(
      backgroundColor = accentColor,
      cornerRadius = 12f,
      modifier = Modifier.horizontalAlignment(Stretch).verticalAlignment(Stretch),
    )
    Column(
      width = Fill,
      horizontalAlignment = Center,
    ) {
      Text(
        "Constraints [x y] | Alignments [x y]",
        modifier = Modifier.horizontalAlignment(Center).height(40.dp),
      )
      Row(
        horizontalAlignment = SpaceBetween,
        verticalAlignment = Center,
      ) {
        Rectangle(
          backgroundColor = boxColor,
          cornerRadius = 4f,
          modifier = Modifier.width(20.dp).height(20.dp),
        )
        Text(" Box")
        Spacer(width = 12.dp)
        Rectangle(
          backgroundColor = backColor,
          cornerRadius = 4f,
          modifier = Modifier.width(20.dp).height(20.dp),
        )
        Text(" Back")
        Spacer(width = 12.dp)
        Rectangle(
          backgroundColor = middleColor,
          cornerRadius = 4f,
          modifier = Modifier.width(20.dp).height(20.dp),
        )
        Text(" Middle")
        Spacer(width = 12.dp)
        Rectangle(
          backgroundColor = frontColor,
          cornerRadius = 4f,
          modifier = Modifier.width(20.dp).height(20.dp),
        )
        Text(" Front")
      }
    }
  }
}

@Composable
private fun ColumnScope.BoxRow(
  width: Constraint,
  height: Constraint,
  horizontalAlignment: CrossAxisAlignment,
  verticalAlignment: CrossAxisAlignment,
  modifier: Modifier = Modifier,
) {
  // Divider
  Rectangle(accentColor, modifier = Modifier.height(1.dp).horizontalAlignment(Stretch).margin(Margin(top = 20.dp)))
  Text(
    "$width $height | $horizontalAlignment $verticalAlignment",
    modifier = Modifier.horizontalAlignment(Center).height(40.dp),
  )
  Column(
    width = Fill,
    height = Fill,
    horizontalAlignment = Start,
    margin = Margin(horizontal = 20.dp, vertical = 0.dp),
    modifier = modifier,
  ) {
    Box(
      width = Fill,
      height = Fill,
      horizontalAlignment = Stretch,
      verticalAlignment = Stretch,
    ) {
      Rectangle(backgroundColor = rowColor)
      Row(
        width = Fill,
        height = Fill,
      ) {
        // This is the box we're measuring.
        Box(
          width = width,
          height = height,
          horizontalAlignment = horizontalAlignment,
          verticalAlignment = verticalAlignment,
          // TODO: Add backgroundColor = boxColor once the modifiers are available.
        ) {
          Rectangle(
            backgroundColor = backColor,
            cornerRadius = 12f,
            modifier = Modifier.width((rowHeight.value * 1.4).dp).height((rowHeight.value * 0.8).dp),
          )
          Rectangle(
            backgroundColor = middleColor,
            cornerRadius = 12f,
            modifier = Modifier.width((rowHeight.value * 1.2).dp).height((rowHeight.value * 0.6).dp),
          )
          Rectangle(
            backgroundColor = frontColor,
            cornerRadius = 12f,
            modifier = Modifier.width((rowHeight.value).dp).height((rowHeight.value * 0.4).dp),
          )
        }
      }
    }
  }

```

### PR DESCRIPTION
**Pull request by JakeWharton**
| Android | iOS |
|---|---|
| Screenshot_20240327_221303 | Simulator Screenshot - iPhone 15 Pro - 2024-03-27 at 22 51 16 |

The behavior of iOS is pretty deeply concerning here. I dramatically reduced the number of containers and explicit sizes in use so that it would behave more dynamically, and as a result iOS completely broke (while Android had basically no change).

Closes #1583 

---

- [ ] `CHANGELOG.md`'s "Unreleased" section has been updated, if applicable.

### CODE REVIEW

**Comment by JakeWharton** [comment]
I did fix the front color in the legend post-screenshot.

File: redwood-layout-view/src/main/kotlin/app/cash/redwood/layout/view/ViewSpacer.kt [hunk 1]
```diff
@@ -28,7 +27,7 @@ internal class ViewSpacer(
 ) : Spacer<View> {
   private val density = Density(context.resources)
 
-  override val value = Space(context)
+  override val value = View(context)
```
Comment (JakeWharton) [inline redwood-layout-view/src/main/kotlin/app/cash/redwood/layout/view/ViewSpacer.kt:30]: `Space` no-ops `draw`, and thus backgrounds wouldn't draw.

File: redwood-layout-view/src/main/kotlin/app/cash/redwood/layout/view/ViewSpacer.kt [hunk 2]
```diff
@@ -17,33 +17,48 @@ package app.cash.redwood.layout.view
...
 import app.cash.redwood.ui.Dp
+import kotlin.math.min
 
 internal class ViewSpacer(
   context: Context,
-) : Spacer<View> {
+) : View(context), Spacer<View> {
   private val density = Density(context.resources)
 
-  override val value = Space(context)
```
Comment (JakeWharton) [inline redwood-layout-view/src/main/kotlin/app/cash/redwood/layout/view/ViewSpacer.kt:31]: `Space` no-ops `draw()`, and thus backgrounds wouldn't draw.

File: test-app/presenter-treehouse/src/commonMain/kotlin/com/example/redwood/testing/treehouse/serializers.kt
Comment (JakeWharton) [inline test-app/presenter-treehouse/src/commonMain/kotlin/com/example/redwood/testing/treehouse/serializers.kt:1]: GG git 🙄

Comment (swankjesse) [reply test-app/presenter-treehouse/src/commonMain/kotlin/com/example/redwood/testing/treehouse/serializers.kt:1]: I think the license headers is what hurts us here. Git’s reasonable algorithm is to find the most similar file, and we’ve got 15 lines of license that cause both files to have more similarity than they should.

**Review by swankjesse — APPROVED** [review]
When I read your concerns i was expecting updated screenshots in the diff. Thankfully it didn’t go that far.

**Comment by swankjesse** [comment]
Rebased!

**Comment by JakeWharton** [comment]
> When I read your concerns i was expecting updated screenshots in the diff. Thankfully it didn’t go that far.

We don't snapshot the test app, but there are tests for box committed that I previously noted seem very broken: https://github.com/cashapp/redwood/pull/1842#discussion_r1511552288. Tracked by #900.

**Pull request merged**